server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2946,13 +2946,7 @@ public UserVm destroyVm(DestroyVMCmd cmd) throws ResourceUnavailableException, C
 
         stopVirtualMachine(vmId, VmDestroyForcestop.value());
 
```

Comment:
@harikrishna-patnala Checked the changes done in the PR https://github.com/apache/cloudstack/pull/4307, which is breaking the old behavior (On destroy VM, detach the data disks which are marked to be deleted, and keep other disks attached). and these changes are reverted here to bring back the old behavior. Am I correct?

Reply:
Yes @sureshanaparti